griptape/loaders/csv_loader.py

```diff
@@ -37,3 +32,20 @@ def _load_file(self, filename: str) -> list[CsvRowArtifact]:
                 artifacts.append(chunk)
 
         return artifacts
+
+    def _stream_from_source(self, source: bytes | str | IO | Path) -> IO:
+        if isinstance(source, bytes):
+            return StringIO(source.decode())
+        elif isinstance(source, str):
+            return StringIO(source)
+        elif isinstance(source, IOBase):
+            return cast(IO, source)
+        elif isinstance(source, Path):
+            return open(source, encoding="utf-8")
```

Comment:
Should this use `self.encoding`?

Reply:
Now I remember why I didn't do this. `CsvLoader` extends `BaseLoader` rather than `BaseTextLoader`, so it doesn't have the `encoding` field.

I feel like it should for usability, so I'll add one and pass it through.